jsonapi/unmarshal.go

```diff
@@ -7,80 +7,77 @@ import (
+// to-many relations.
 type UnmarshalToManyRelations interface {
 	SetToManyReferenceIDs(name string, IDs []string) error
 }
 
-// The EditToManyRelations interface can be optionally implemented to add and delete to-many
-// relationships on a already unmarshalled struct. These methods are used by our API for the to-many
-// relationship update routes.
-/*
-There are 3 HTTP Methods to edit to-many relations:
-
-	PATCH /v1/posts/1/comments
-	Content-Type: application/vnd.api+json
-	Accept: application/vnd.api+json
-
-	{
-	  "data": [
-		{ "type": "comments", "id": "2" },
-		{ "type": "comments", "id": "3" }
-	  ]
-	}
-
-this replaces all of the comments that belong to post with ID 1 and the SetToManyReferenceIDs method
-will be called
-
-	POST /v1/posts/1/comments
-	Content-Type: application/vnd.api+json
-	Accept: application/vnd.api+json
-
-	{
-	  "data": [
-		{ "type": "comments", "id": "123" }
-	  ]
-	}
-
-adds a new comment to the post with ID 1. The AddToManyIDs methid will be called.
-
-	DELETE /v1/posts/1/comments
-	Content-Type: application/vnd.api+json
-	Accept: application/vnd.api+json
-
-	{
-	  "data": [
-		{ "type": "comments", "id": "12" },
-		{ "type": "comments", "id": "13" }
-	  ]
-	}
-
-deletes comments that belong to post with ID 1. The DeleteToManyIDs method will be called.
-*/
+// The EditToManyRelations interface can be optionally implemented to add and
+// delete to-many relationships on a already unmarshalled struct. These methods
+// are used by our API for the to-many relationship update routes.
+//
+// There are 3 HTTP Methods to edit to-many relations:
+//
+//	PATCH /v1/posts/1/comments
+//	Content-Type: application/vnd.api+json
+//	Accept: application/vnd.api+json
+//
+//	{
+//	  "data": [
+//		{ "type": "comments", "id": "2" },
+//		{ "type": "comments", "id": "3" }
+//	  ]
+//	}
+//
+// This replaces all of the comments that belong to post with ID 1 and the
+// SetToManyReferenceIDs method will be called.
+//
+//	POST /v1/posts/1/comments
+//	Content-Type: application/vnd.api+json
+//	Accept: application/vnd.api+json
+//
+//	{
+//	  "data": [
+//		{ "type": "comments", "id": "123" }
+//	  ]
+//	}
+//
+// Adds a new comment to the post with ID 1.
+// The AddToManyIDs method will be called.
+//
+//	DELETE /v1/posts/1/comments
+//	Content-Type: application/vnd.api+json
+//	Accept: application/vnd.api+json
+//
+//	{
+//	  "data": [
+//		{ "type": "comments", "id": "12" },
+//		{ "type": "comments", "id": "13" }
+//	  ]
+//	}
+//
+// Deletes comments that belong to post with ID 1.
+// The DeleteToManyIDs method will be called.
 type EditToManyRelations interface {
 	AddToManyIDs(name string, IDs []string) error
 	DeleteToManyIDs(name string, IDs []string) error
 }
 
-var (
-	// ErrInterface occurs if the target struct did not implement the UnmarshalIdentifier interface
-	ErrInterface = errors.New("target must implement UnmarshalIdentifier interface")
```

Comment:
Hm, we made this a variable and exported from the package because we wanted that somebody from the outside is able to check against this error variable. But I guess for this error it does not really matter. However for the `ErrNoType` it could be useful right? Because that actually checks that the `type` field is missing int he JSON that needs to be unmarshaled.